### PR TITLE
chore/test-operance-data-column-width

### PR DESCRIPTION
**Pull request by zxcvbnmmohd**
- write full tests for OperanceDataColumnWidth class
  - update lefthook to include generation of tests in pre-commit hook
  - add test coverage report

### CODE REVIEW

File: lefthook.yml [hunk 1]
```diff
@@ -13,14 +13,19 @@ pre-commit:
...
+        coverage_percentage=$(lcov --summary coverage/lcov.info | grep -Po '\d+(?=%)' | head -1)
+        if [ "$coverage_percentage" -ne 100 ]; then
+          echo "ERROR: Test coverage is not 100%. Current coverage is $coverage_percentage%."
+          exit 1
+        fi
+        git add .
 
 pre-push:
   commands:
-    full-lint:
```
Comment (zxcvbnmmohd) [inline lefthook.yml:20]: don't really need this as it's being done above on only the files that's been changed

File: lib/src/operance_data_column_width.dart [hunk 2]
```diff
@@ -11,7 +11,7 @@ class OperanceDataColumnWidth {
     this.factor = 0.15,
   }) : assert(
           factor > 0 && factor <= 1,
-          'Factor must be between 0 and 1',
+          'factor must be between 0 and 1',
```
Comment (zxcvbnmmohd) [inline lib/src/operance_data_column_width.dart:14]: slipped in from before, the messages should contain the variable name as is

File: test/operance_data_column_width_test.dart
Comment (zxcvbnmmohd) [inline test/operance_data_column_width_test.dart:1]: This is the structure of the tests I expect, the Given, When, Then format of tests

Comment (zxcvbnmmohd) [inline test/operance_data_column_width_test.dart:1]: This is the structure of the tests I expect: the Given ..., When ..., Then ... format of tests.

**Comment by zxcvbnmmohd** [comment]
Anything within the coverage folder is generated when you commit your work so ignore those

File: lefthook.yml (hunk 1 above)
Comment (fjyeo) [reply lefthook.yml:20]: Yep this makes sense

File: lib/src/operance_data_column_width.dart [hunk 3]
```diff
@@ -11,7 +11,7 @@ class OperanceDataColumnWidth {
     this.factor = 0.15,
   }) : assert(
           factor > 0 && factor <= 1,
```
Comment (fjyeo) [inline lib/src/operance_data_column_width.dart:13]: The assertion could be modified to only check the factor when size is null. This allows for more flexibility in setting the factor when a fixed size is provided.

```suggestion

}) : assert(
          size != null || (factor > 0 && factor <= 1),
          'factor must be between 0 and 1 when size is not provided',
        );


```

File: test/operance_data_column_width_test.dart
Comment (fjyeo) [inline test/operance_data_column_width_test.dart:1]: This is good. To be extra thorough we could add a few more tests for example test with a very small factor (0.001) + test with a factor very close to 1 and test with various total widths to ensure consinstent behaviour but overall great.

**Review by fjyeo — COMMENTED** [review]
I think it's all good just one change I would make and some extra tests for us to think about.

**Review by fjyeo — APPROVED** [review]
Files are all great. Good job 👍

**Comment by fjyeo** [comment]
Let's get it in then

**Pull request merged**